### PR TITLE
[E-idora-create < T458-MAGE] Implement remove labels for create functions

### PR DESCRIPTION
**Pull request by ind1xa**
### Description

Implemented create.remove_labels and e2e tests for it

### Pull request type

- [ ] Bugfix
- [x ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/create_module/algorithm/create.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          throw mgp::ValueException("First argument must be a node, node's id or a list of those.");
+        }
+      }
+    } else if (arguments[0].IsNode()) {
+      nodeIds.insert(std::move(arguments[0].ValueNode().Id()));
+    } else if (arguments[0].IsInt()) {
+      nodeIds.insert(std::move(mgp::Id::FromInt(arguments[0].ValueInt())));
+    } else {
+      throw mgp::ValueException("First argument must be a node, node's id or a list of those.");
+    }
```
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:28]: Can you isolate this in function so it accepts mgp::Value?

**Review by antoniofilipovic — COMMENTED** [review]
One discussion point

File: cpp/create_module/algorithm/create.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  } else if (argument.IsNode()) {
+    result_set.insert(std::move(argument.ValueNode().Id()));
+  } else if (argument.IsInt()) {
+    result_set.insert(std::move(mgp::Id::FromInt(argument.ValueInt())));
+  } else {
+    throw mgp::ValueException("First argument must be a node, node's id or a list of those.");
+  }
+  return result_set;
+}
```
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:23]: You can change this according to other function in PR: https://github.com/memgraph/mage/pull/273

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Refactor function and it should be good to go

File: cpp/create_module/algorithm/create.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    auto node = element.ValueNode();
+    RemoveElementLabels(node, labels, record_factory);
+    return;
+  }
+  if (element.IsInt()) {
+    auto node = graph.GetNodeById(mgp::Id::FromInt(element.ValueInt()));
+    RemoveElementLabels(node, labels, record_factory);
+    return;
+  }
+  throw mgp::ValueException("First argument must be type node, relationship, id or a list of those.");
```
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:24]: Add this check before other checks, so you can have this order.

pseudocode
```
check if not correct type ():
  throw errro // this is early return part 1
check if type node():
  do something...
  return // early return part 2
do other if int...
```

File: cpp/create_module/algorithm/create.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+void Create::ProcessElement(const mgp::Value &element, const mgp::Graph graph, const mgp::List &labels,
+                            const mgp::RecordFactory &record_factory) {
+  if (element.IsNode()) {
+    auto node = element.ValueNode();
+    RemoveElementLabels(node, labels, record_factory);
+    return;
+  }
+  if (element.IsInt()) {
```
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:19]: Can this int here be relationship id?

File: cpp/create_module/algorithm/create.cpp (hunk 3 above)
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:24]: Where are we processing the relationship?

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
One suggestion and few discussion points

File: cpp/create_module/algorithm/create.cpp (hunk 4 above)
Comment (ind1xa) [reply cpp/create_module/algorithm/create.cpp:19]: No it can't, sorry I copied the exception message from other procedure and forgot to change it

**Pull request merged**